datafusion/core/src/datasource/physical_plan/parquet/mod.rs

```diff
@@ -586,12 +662,8 @@ impl ParquetExec {
         )
     }
 
-    fn with_file_groups(mut self, file_groups: Vec<Vec<PartitionedFile>>) -> Self {
-        self.base_config.file_groups = file_groups;
-        // Changing file groups may invalidate output partitioning. Update it also
-        let output_partitioning = Self::output_partitioning_helper(&self.base_config);
-        self.cache = self.cache.with_partitioning(output_partitioning);
-        self
+    fn with_file_groups(self, file_groups: Vec<Vec<PartitionedFile>>) -> Self {
+        self.into_builder().with_file_groups(file_groups).build()
```

Comment:
the builder forces the recalculation of the various properties -- I think the existing code actually has a bug that the statistics are not recomputed when new file groups are added

Reply:
update-- I was wrong. The existing code only uses the existing file groups so statistics don't need to be updated. I renamed this function to make this more clear